app/jobs/bulkrax/child_relationships_job.rb

```diff
@@ -0,0 +1,97 @@
+module Bulkrax
+  class ChildWorksError < Exception; end
+  class ChildRelationshipsJob < ApplicationJob
+    queue_as :import
+
+    def perform(*args)
+      begin
+        @entry = Entry.find(args[0])
+        @child_entries = args[1].map { |e| Entry.find(e) }
+        @child_works_hash = build_child_works_hash
+        @importer_run_id = args[2]
+        @user = @entry.importerexporter.user
+
+        if (@entry.factory_class == Collection)
+          # add collection to works
+          member_of_collection = []
+          @child_works_hash.each_pair { |k,v| member_of_collection << k if v[:class_name] != 'Collection' }
+          member_of_collection.each { |work| work_child_collection_parent(work) }
+
+          # add collections to collection
+          members_collections = []
+          @child_works_hash.each_pair { |k,v| members_collections << k if v[:class_name] == 'Collection' }
+          collection_parent_collection_child(members_collections) unless members_collections.blank?
+        else
+          # add works to work
+          members_works = []
+          # reject any Collections, they can't be children of Works
+          @child_works_hash.each_pair {|k,v| members_works << k if v[:class_name] != 'Collection' }
+          if members_works.length < @child_entries.length
+            Rails.logger.warn("Cannot add collections as children of works: #{(@child_entries.length - members_works.length)} collections were discarded for parent entry #{@entry.id} (of #{@child_entries.length})")
+          end
+          work_parent_work_child(members_works) unless members_works.blank?
+        end
+      # Not all of the Works/Collections exist yet; reschedule
+      rescue Bulkrax::ChildWorksError
+        reschedule(args[0], args[1], args[2])
+      # Exceptions here are not an issue with adding the relationships.
+      # Those are caught seperately, these are more likely network, db or other unexpected issues.
+      # Note that these temporary type issues do not raise the failure count
+      rescue StandardError => e
```

Comment:
this code is redudenant. if you remove `rescue StandardError => e; raise e` it will do exactly the same thing